### DISCUSSION TITLE
How to copy widgets to another bar on another screen?

### DISCUSSION THREAD

**Discussion by ervinpopescu** [qtile/qtile#4431] [Q&A] [answered]
```py
widgets_2 = []
for w in widgets_1:
    if not isinstance(w, (widget.Systray, widget.TaskList)):
        copied = copy.copy(w)
    if isinstance(w, widget.TaskList):
        copied = copy.deepcopy(w)
    widgets_2.append(copied)
```

I would especially like to have a TaskList on each Screen.

**Comment by elParaguayo** [comment]
If you want exactly the same widget on each screen then create an instance of the widget and then put the same instance in both screens. Qtile will convert one of these to a `Mirror` which "reflects" the contents of the other one.

You can't do this with `Systray`.

**Reply by ervinpopescu** [reply]
Oof, misspoke again, sorry! I know the `Systray` issue, I have solved it in the above code, but the issue I'm having is that `TaskList`s are mirrored to the other `Screen` instead of being independent of each other and showing only the windows that are currently open on their corresponding `Screen`.  The above approach doesn't work, with no error in the logs AFAICT, because it doesn't add the `TaskList` to the other `Screen`. That is why I was trying to `deepcopy` the instance into another one, but I am not familiar with that, and have no idea why it's failing.

Diff of the widgets: https://termbin.com/0o1g

**Reply by elParaguayo** [reply]
Looking at that diff, you're getting a mirror, not a separate widget. Therefore it's going to show the same contents. Instead of copying it, you need to create a new instance of the widget.

**Reply by ervinpopescu** [reply]
As you can see in [widgets](https://github.com/ervinpopescu/dots/tree/lenovo/configs/.config/qtile/modules/widgets) I've got a different file/module for each widget, as I've made a poor decision at some point at my life. Therefore, I have been relying on mirroring this far without realizing this issue, as I haven't really used another screen that often. I would still like to know if there is any possibility of *not* creating a new instance, but reusing the old ones, by copying or some other pythonic operation. Thanks!

**Reply by elParaguayo** [reply]
I would have expected `copy.copy` to work. A mirror is created when the bar tries to configure a widget that has already been configured (the `configured` attribute is set to `True`). This would happen if the same instance is in multiple bars but _shouldn't_ be the case where one bar has a copy of the other.

**Reply by ervinpopescu** [reply, marked as answer]
It works!
```py
widgets_2 = []
for w in widgets_1:
    if not isinstance(w, (widget.Systray, WidgetBox)):
        copied = copy.copy(w)
        widgets_2.append(copied)
```
Thanks a lot!
